src/fairseq2/datasets/nllb.py

```diff
@@ -0,0 +1,409 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from __future__ import annotations
+
+import logging
+from functools import partial
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Sequence, Tuple, TypedDict, cast, final
```

Comment:
nit but we can avoid importing `Dict` etc (since you are importing `annotations`) and use `dict` instead e.g.

Reply:
I just want to be consistent throughout the code base. Since we still support Python 3.8 where some of the newer type annotation syntax is no supported, I prefer to keep everything "compatible" with Python 3.8.